doc/shared/sundomeigest/SUNDomEigEst_API.rst

```diff
@@ -0,0 +1,622 @@
+   +------------------------------------+-------+---------------------------------------------------+
+
+
+.. _SUNDomEigEst.Generic:
+
+The generic SUNDomEigEstimator module
+-----------------------------------------
+
+SUNDIALS packages interact with dominant eigenvalue estimator implementations through the
+:c:type:`SUNDomEigEstimator` class.  A :c:type:`SUNDomEigEstimator` is a pointer to the
+:c:struct:`_generic_SUNDomEigEstimator` structure:
+
+.. c:type:: struct _generic_SUNDomEigEstimator *SUNDomEigEstimator
+
+.. c:struct:: _generic_SUNDomEigEstimator
+
+   The structure defining the SUNDIALS dominant eigenvalue estimator class.
+
+   .. c:member:: void *content
+
+      Pointer to the dominant eigenvalue estimator-specific member data
+
+   .. c:member:: SUNDomEigEstimator_Ops ops
+
+      A virtual table of dominant eigenvalue estimator operations provided by a specific
+      implementation
+
+   .. c:member:: SUNContext sunctx
+
+      The SUNDIALS simulation context
+
+The virtual table structure is defined as
+
+.. c:type:: struct _generic_SUNDomEigEstimator_Ops *SUNDomEigEstimator_Ops
+
+.. c:struct:: _generic_SUNDomEigEstimator_Ops
+
+   The structure defining :c:type:`SUNDomEigEstimator` operations.
+
+   .. c:member:: SUNErrCode (*setatimes)(SUNDomEigEstimator, void*, SUNATimesFn)
+
+      The function implementing :c:func:`SUNDomEigEst_SetATimes`
+
+   .. c:member:: SUNErrCode (*setmaxiters)(SUNDomEigEstimator, int)
+
+      The function implementing :c:func:`SUNDomEigEst_SetMaxIters`
+
+   .. c:member:: SUNErrCode (*setnumpreprocess)(SUNDomEigEstimator, int)
+
+      The function implementing :c:func:`SUNDomEigEst_SetNumPreProcess`
+
+   .. c:member:: SUNErrCode (*settol)(SUNDomEigEstimator, sunrealtype)
+
+      The function implementing :c:func:`SUNDomEigEst_SetTol`
+
+   .. c:member:: SUNErrCode (*initialize)(SUNDomEigEstimator)
+
+      The function implementing :c:func:`SUNDomEigEst_Initialize`
+
+   .. c:member:: SUNErrCode (*estimate)(SUNDomEigEstimator, sunrealtype*, sunrealtype*)
+
+      The function implementing :c:func:`SUNDomEig_Estimate`
+
+   .. c:member:: sunrealtype (*getcurres)(SUNDomEigEstimator)
+
+      The function implementing :c:func:`SUNDomEigEst_GetCurRes`
+
+   .. c:member:: int (*getcurniters)(SUNDomEigEstimator)
+
+      The function implementing :c:func:`SUNDomEigEst_GetCurNumIters`
+
+   .. c:member:: int (*getmaxniters)(SUNDomEigEstimator)
+
+      The function implementing :c:func:`SUNDomEigEst_GetMaxNumIters`
+
+   .. c:member:: int (*getminniters)(SUNDomEigEstimator)
+
+      The function implementing :c:func:`SUNDomEigEst_GetMinNumIters`
+
+   .. c:member:: long int (*getnumatimescalls)(SUNDomEigEstimator)
+
+      The function implementing :c:func:`SUNDomEigEst_GetNumATimesCalls`
+      
+   .. c:member:: SUNErrCode (*printstats)(SUNDomEigEstimator, FILE*)
+
+      The function implementing :c:func:`SUNDomEigEst_PrintStats`
+      
+   .. c:member:: SUNErrCode (*free)(SUNDomEigEstimator)
```

Comment:
Typically, the function pointer name matches the public API name, and this forgot a `*`
```suggestion
   .. c:member:: SUNErrCode (*destroy)(SUNDomEigEstimator*)
```